app/src/renderer/vuex/modules/blockchain.js

```diff
@@ -69,21 +70,39 @@ export default ({ commit, node }) => {
       blockMetaInfo && state.blockMetas.push(blockMetaInfo)
       return blockMetaInfo
     },
-    subscribeToBlocks ({ commit }) {
-      node.rpc.subscribe({ query: "tm.event = 'NewBlock'" }, (err, event) => {
-        state.subscription = true
+    subscribeToBlocks ({ state, commit, dispatch }) {
+      // ensure we never subscribe twice
+      if (state.subscription) return
 
-        if (err) {
+      function error (err) {
+        state.subscription = false
+        commit('notifyError', { title: `Error subscribing to new blocks`, body: err.message })
+      }
+
+      node.rpc.status((err, status) => {
+        if (err) return error(err)
+
+        if (status.syncing) {
+          // still syncing, let's try subscribing again in 30 seconds
+          state.syncing = true
           state.subscription = false
-          commit('notifyError', { title: `Error subscribing to new blocks`, body: err.message })
+          setTimeout(() => dispatch('subscribeToBlocks'), 30e3)
```

Comment:
30e3? never seen that before. can't find anything on google about it.

Reply:
Scientific notation for 30000, maybe I should do `30 * 1000` so it's more obvious.